stable/yugabyte/templates/setup-credentials-configmap.yaml

```diff
@@ -0,0 +1,259 @@
+{{- if or .Values.authCredentials.ycql.user .Values.authCredentials.ycql.password .Values.authCredentials.ycql.keyspace .Values.authCredentials.ysql.password .Values.authCredentials.ysql.user .Values.authCredentials.ysql.database }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "yugabyte.fullname" . }}-setup-credentials-script
+  namespace: "{{ .Release.Namespace }}"
+  labels:
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}"
+    component: "{{ .Values.Component }}"
+data:
+  setup-credentials.sh: |
+    #!/bin/bash
+
+    set -eo pipefail
+
+    # Setup script to setup credentials
+
+    # -----------------------------------------
+    # Default Variables
+    # -----------------------------------------
+
+    readonly DEFAULT_YSQL_USER="yugabyte"
+    readonly DEFAULT_YSQL_PASSWORD="yugabyte"
+    readonly DEFAULT_YSQL_DB="yugabyte"
+
+    readonly DEFAULT_YCQL_USER="cassandra"
+    readonly DEFAULT_YCQL_PASSWORD="cassandra"
+
+    {{- range .Values.Services }}
+    {{- $service := . -}}
+      {{- if eq ($service.name) "yb-tservers" }}
+        readonly YSQL_PORT={{ index $service.ports "tcp-ysql-port" }}
+        # TODO: Update the tcp-yql-port to tcp-ycql-port in values.yaml
+        readonly YCQL_PORT={{ index $service.ports "tcp-yql-port" }}
+      {{- end }}
+    {{- end }}
+
+    prefix_ysql_cmd=(
+      /home/yugabyte/bin/ysqlsh -h yb-tservers.{{ .Release.Namespace }}.svc.{{ .Values.domainName }}
+      -p "$YSQL_PORT"
+    )
+
+    prefix_ycql_cmd=(
+      /home/yugabyte/bin/ycqlsh yb-tservers.{{ .Release.Namespace }}.svc.{{ .Values.domainName }}
+      "$YCQL_PORT"
+    )
+
+    {{- if .Values.tls.enabled }}
+      prefix_ysql_cmd+=("sslmode=require")
+      prefix_ycql_cmd+=(--ssl)
+    {{- end }}
+
+    # -----------------------------------------
+    # Variables
+    # -----------------------------------------
+
+    ysql_user=
+    ysql_password=
+    ysql_db=
+    ycql_user=
+    ycql_password=
+    ycql_keyspace=
+
+    # -----------------------------------------
+    # Hepler functions
+    # -----------------------------------------
+
+    cleanup() {
+      local exit_code=$?
+      echo "Exiting with code $exit_code"
+      exit "$exit_code"
+    }
+
+    function waitUntilHealthy() {
+      declare -a ysql_cmd
+      export PGPASSWORD="$2"
+      ysql_cmd=(
+        /home/yugabyte/bin/ysqlsh -h yb-tservers.{{ .Release.Namespace }}.svc.{{ .Values.domainName }}
+        -p "$3"
+        -U "$1"
+        -c "\\conninfo"
+      )
+
+      if [[ "$4" == "true" ]]; then
+        ysql_cmd+=("sslmode=require")
+      fi
+
+      echo "${ysql_cmd[@]}"
+      while ! "${ysql_cmd[@]}"; do
+        sleep 5s
```

Comment:
log is already added at line no. 217.

This part exists in the `waitUntilHealthy` function and, this function is used with a timeout of 3m at line no.218. The script will wait for 3 mins to get the cluster healthy, but somehow the cluster is unhealthy then the script exit with status 1. And the "backoffLimit" is 2 so K8s will execute the hook two times in case of failure and even after 2 executions hook failed then the `kubectl/helm` exit and return the error something like "post hook failed". We're removing the post-install hook job pod only on success so in case of failure, we can check the logs of the job and can do some debugging.